lib/infrastructure/referral.repository.dart

```diff
@@ -0,0 +1,77 @@
+import 'package:clones_desktop/domain/models/api/create_referral_request.dart';
```

Comment:
Normally, you should have some warning due to `analysis_options.yaml` rules
For example in this class:
- Unused import
- Sort directive sections alphabetically
- 'ReferralRepositoryRef' is deprecated and shouldn't be used. Will be removed in 3.0. Use Ref instead.
- Unnecessary use of a 'double' literal.
- The operand can't be 'null', so the condition is always 'false'.
- The value of the argument is redundant 
- Missing a newline at the end of the file.